From 08049815698675363a57fdde2173441750852914 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Wed, 10 Oct 2018 07:06:28 -0600
Subject: [PATCH 3/5] msm8974-common: Set TARGET_NEEDS_NETD_DIRECT_CONNECT_RULE
 to true

* Due to "specialness" in our RIL library, ConnectivityService
  doesn't seem to appropriately add the relevant directly-
  connected routes, meaning that IPv4 mobile data is inoperable
  due to lack of a default route. Let's reenable the legacy netd
  behavior to get our connectivity back.

Change-Id: I59f8476ab36db0fb79856f74346e62e0dd4abae0
---
 BoardConfigCommon.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 6ef0a37..204e5ca 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -88,6 +88,9 @@ include device/samsung/msm8974-common/sepolicy/sepolicy.mk
 TARGET_INIT_VENDOR_LIB := libinit_msm8974
 TARGET_RECOVERY_DEVICE_MODULES := libinit_msm8974
 
+# Network Routing
+TARGET_NEEDS_NETD_DIRECT_CONNECT_RULE := true
+
 # Power
 TARGET_HAS_LEGACY_POWER_STATS := true
 TARGET_HAS_NO_WLAN_STATS := true
-- 
2.17.1

